sdcm/tester.py

```diff
@@ -2584,49 +2599,91 @@ def copy_data_between_tables(self, node, src_keyspace, src_table, dest_keyspace,
         self.log.debug('All rows have been copied from %s to %s', src_table, dest_table)
         return True
 
-    def collect_partitions_info(self, table_name, primary_key_column, save_into_file_name) -> dict | None:  # pylint: disable=too-many-locals
+    def validate_rows_per_partitions(self, skip_on_rows_number_limit: bool = True):
+        """
+        Validating partition rows-number is the same before and after running a nemesis/stress.
+        The purpose of "skip_on_rows_number_limit" is to avoid a "too heavy" scan in a too often occurrence,
+        e.g. every health check.
+        For example, if rows-per-partition is 10M and there are 50 partitions for validation,
+        It would be too much of an overhead to read 500M rows every time a nemesis is completed and health-check
+        is running. so, by default, it is limited to LIMIT_TOTAL_ROWS_NUMBER.
+        """
+        if not (self.validate_partitions and self.partitions_attributes and self.partitions_dict_before):
+            self.log.debug('Skipping validate partitions info')
+            return
+
+        if skip_on_rows_number_limit:
+            # estimated_total_rows_number represents number-of-partitions * number-of-rows-per-partition
+            # Meaning the total number of estimated rows to be read.
+            estimated_total_rows_number = len(self.partitions_dict_before) * \
+                int(list(self.partitions_dict_before.values())[0])
+            if estimated_total_rows_number > TOTAL_ROWS_NUMBER_LIMIT:
+                self.log.debug(
+                    'The estimated total number of rows (%s) exceeds allowed limit (%s). Skipping validation.',
+                    estimated_total_rows_number, TOTAL_ROWS_NUMBER_LIMIT)
+                return
```

Comment:
so, if the `estimated_total_rows_number` is greater than `TOTAL_ROWS_NUMBER_LIMIT` we are just not runnin `self.collect_partitions_info()` as the function description suggests...
should we then assume that in this case the number of lines to run would be the `TOTAL_ROWS_NUMBER_LIMIT`?
if that is the case, then this `return` is wrong, and we need to assume the max number of rows to run.

Reply:
redesigned, removed.